docs/topics/http/urls.txt

```diff
@@ -597,16 +597,34 @@ In order to perform URL reversing, you'll need to use **named URL patterns**
 as done in the examples above. The string used for the URL name can contain any
 characters you like. You are not restricted to valid Python names.
 
-When you name your URL patterns, make sure you use names that are unlikely
-to clash with any other application's choice of names. If you call your URL
-pattern ``comment``, and another application does the same thing, there's
-no guarantee which URL will be inserted into your template when you use
-this name.
+First, when you name your own URL patterns, make sure you use names that are
+unlikely to clash with any other application's choice of names. If you call
+your URL pattern ``comment``, and another application does the same thing,
+there's no guarantee which URL will be inserted into your template when you
+use this name, it will depend on the order of .
 
 Putting a prefix on your URL names, perhaps derived from the application
 name, will decrease the chances of collision. We recommend something like
 ``myapp-comment`` instead of ``comment``.
 
+Second, you can deliberately use the *same name* as used in another (built-in)
+application. A common usage is to override the built-in login view
+(:class:`~django.contrib.auth.views.LoginView`). Django gives that view a URL
+pattern with the name ``login``. If you have a custom login view and give its
+URL the name ``login``, most login actions in other applications will
+automatically use your custom view.
+
+Third, you can use the same name for multiple URL patterns if they differ
+regarding (keyword) arguments. When looking up URL patterns by name, django
+makes sure the number of arguments and the names of the keyword arguments
+match.
+
+A note on order. URL patterns are matched from the top down: the first pattern
+to match is used. URL pattern names are looked up in reverse order: from the
+bottom to the top. So a named url further down in your URL conf wins from one
+defined earlier. All URLs in your URL conf come before any included URLs,
```

Comment:
I'm not positive about what this means. I think it might mean "All `url()`s in your ROOT_URLCONF take precedence over `url()`s from `include()`s in ROOT_URLCONF." Is it correct? This seems a bit unintuitive and might merit some more explanation and tests.